tools/purge_ems.rb

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env ruby
+enviroment = File.expand_path('../config/environment', __dir__) + ".rb"
+require enviroment if File.exist?(enviroment)
+
+if !defined?(Vmdb::Application)
+  warn "This script must exist in the tools directory or can live elsewhere but needs to be run via rails runner path/to/purge_ems.rb --id 1234 --follow true"
+  exit 1
+end
+
+require 'optimist'
+
+ARGV.shift if ARGV[0] == '--' # if invoked with rails runner
+opts = Optimist.options do
+  banner "Purge managements system records in the background via the queue.\n\nUsage: ruby #{$0} [options]\n\nOptions:\n\t"
+  opt :id,      "Mangement System id",                                         :short => :i, :type => :integer,                    :required => true
+  opt :follow,  "Follow the progress or exit after creating messages",         :short => :f, :type => :string,  :default => "true"
+  opt :batch,   "How many rows in each relationship to be deleted in a batch", :short => :b, :type => :integer, :default => 1_000
+  opt :timeout, "How many minutes should each message be allowed to run",      :short => :t, :type => :integer, :default => 30
+end
+
+Optimist.die :id,      "must be a positive number"                             if opts[:id] < 1
+Optimist.die :batch,   "must be a positive number"                             if opts[:batch] < 1
+Optimist.die :timeout, "must be a positive number greater than or equal to 10" if opts[:timeout] < 10
+
+opts[:follow] = opts[:follow].to_s.downcase
+Optimist.die :follow, "must be true or false" unless %w[true false].include?(opts[:follow])
+
+RELATIONSHIP_DESTROY_METHOD = "destroy".freeze
+EMS_DESTROY_METHOD          = "destroy_queue".freeze
+ZONE                        = nil                    # Will queue for any zone
+PRIORITY                    = MiqQueue::LOW_PRIORITY # Don't block other work
```

Comment:
Thought/concerns on making it low priority?  My thoughts were we could be dumping hundreds of messages on the queue and anything added by the scheduler or anything else at normal priority would sit behind this work which could take a while.  The risk is we never run our stuff. 🤣